src/components/plots/barPlot.ts

```diff
@@ -16,6 +19,9 @@ export module Plots {
     private _labelFormatter: Formatter = Formatters.identity();
     private _labelsEnabled = false;
     private _hideBarsIfAnyAreTooWide = true;
+    private _labelAreas: Utils.Map<Dataset, D3.Selection>;
+    private _labelMeasurers: Utils.Map<Dataset, SVGTypewriter.Measurers.Measurer>;
+    private _labelWriters: Utils.Map<Dataset, SVGTypewriter.Writers.Writer>;
```

Comment:
From an instance variable point of view, yeah.  From an implementation point of view, it's 1 step more trouble unless we have a good semantic way to group area / measurer / writer.